### PR TITLE
WIP: Add `--stdin` option to lint from stdin

### PR DESCRIPTION
**Pull request by adtac**
### Checklist

- [x] I have rebased properly. Please see [our tutorial on rebasing](http://coala.readthedocs.io/en/latest/Developers/Git_Basics.html#rebasing).
- [x] I have gone through the [commit guidelines](http://coala.readthedocs.io/en/latest/Developers/Writing_Good_Commits.html) and I've followed them.
- [x] I have followed the [guidelines for the commit shortlog](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#shortlog).
- [x] I have followed the [guidelines for the commit body](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#commit-body).
- [x] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#issue-reference).
- [x] I have [run all the tests](http://coala.readthedocs.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!

### CODE REVIEW

**Comment by Mixih** [comment]
@adtac rebasing should clear all of the build errors

**Comment by Mixih** [comment]
@adtac looks like you need to write a test for your new function `mode_stdin`

File: coalib/coala_modes.py [hunk 1]
```diff
@@ -92,3 +92,36 @@ def mode_format():
     _, exitcode, _ = run_coala(
             print_results=print_results_formatted)
     return exitcode
+
+
+def mode_stdin(console_printer, log_printer):
+    import functools
```
Comment (Mixih) [inline coalib/coala_modes.py]: Needs a test. Probably feeding it args and verifying the exitcode is sufficient.

File: coalib/coala.py [hunk 2]
```diff
@@ -77,6 +78,9 @@ def main():
     if args.format:
         return mode_format()
 
+    if args.stdin:
+        return mode_stdin(console_printer, log_printer)
```
Comment (Mixih) [inline coalib/coala.py]: Might need test

**Comment by adtac** [comment]
@Mixih ah right, forgot :+1:

**Comment by adtac** [comment]
This is quite outdated :)

**Pull request closed without merging**